controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -523,7 +632,7 @@ public String getScopeName() {
 
     @Override
     public CompletableFuture<TxnStatus> abortTransaction(final int epoch, final UUID txId) {
-        CompletableFuture<TxnStatus> future = verifyLegalState(checkTransactionStatus(txId).thenApply(x -> {
+        CompletableFuture<TxnStatus> future = verifyLegalState(() -> checkTransactionStatus(txId).thenApply(x -> {
```

Comment:
Same here: This method shall be called only if the transaction is created and then subsequently sealed. Do we really need `verifyLegalState` check here. I think it is redundant.